chunky/src/java/se/llbit/math/Octree.java

```diff
@@ -46,22 +45,31 @@
     Node get(int x, int y, int z);
     Material getMaterial(int x, int y, int z, BlockPalette palette);
     void store(DataOutputStream output) throws IOException;
-    boolean isInside(Vector3 pos);
+    int getDepth();
+    long nodeCount();
+    NodeId getRoot();
+    boolean isBranch(NodeId node);
+    NodeId getChild(NodeId parent, int childNo);
+    int getType(NodeId node);
+    int getData(NodeId node);
+  }
 
-    /**
-     * Intersects the ray with the geometry inside the octree.
-     *
-     * @return {@code false} if the ray did not hit the geometry
-     */
-    boolean enterBlock(Scene scene, Ray ray, BlockPalette palette);
+  public interface NodeId {}
```

Comment:
an empty interface?

Reply:
It just exist to allow octree to manipulate it, it is given to the implementation when information is needed so the method to handle that are in the implementation

Reply:
Oh, allright! :+1: